### PR TITLE
Add MP JWT TCK

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

File: extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallRyeJwtProcessor.java [hunk 1]
```diff
@@ -47,7 +69,7 @@ void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBe
         removable.addBeanClass(JsonValueProducer.class);
         removable.addBeanClass(JwtPrincipalProducer.class);
         additionalBeans.produce(removable.build());
-
+        additionalBeans.produce(new AdditionalBeanBuildItem(Claim.class));
```
Comment (mkouba) [inline extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallRyeJwtProcessor.java]: `AdditionalBeanBuildItem` is actually removable by default, so you can just add `Claim` to the `removable` build item above...

File: test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusDeployableContainer.java [hunk 2]
```diff
@@ -198,7 +199,10 @@ public void execute(BuildContext context) {
 
         ProtocolMetaData metadata = new ProtocolMetaData();
         URI uri = URI.create(TestHTTPResourceManager.getUri());
-        metadata.addContext(new HTTPContext(uri.getHost(), uri.getPort()));
+        HTTPContext httpContext = new HTTPContext(uri.getHost(), uri.getPort());
+        // This is to work around https://github.com/arquillian/arquillian-core/issues/216
+        httpContext.add(new Servlet("dummy", "/"));
```
Comment (mkouba) [inline test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusDeployableContainer.java]: Nice one :D

**Review by mkouba — COMMENTED** [review]
Otherwise looks good!

**Review by mkouba — APPROVED** [review]

File: extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallRyeJwtProcessor.java (hunk 1 above)
Comment (kenfinnigan) [reply extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallRyeJwtProcessor.java]: Fixed

**Comment by sberyozkin** [comment]
@kenfinnigan thanks for wiring it in

**Comment by kenfinnigan** [comment]
Not sure why native REST Client test fails with these changes, didn't on a previous run

**Comment by kenfinnigan** [comment]
Does anyone have Azure access to re-run the Native Tests main job?

**Comment by mkouba** [comment]
@kenfinnigan pls rebase - I've seen similar issues yesterday. I'll re-run the CI afterwards...

**Comment by kenfinnigan** [comment]
Just rebased

**Comment by kenfinnigan** [comment]
oh come on, those two builds worked last time!

**Pull request merged**